### ISSUE TITLE
local variable 'lipid_details' referenced before assignment

### ISSUE THREAD

**Issue by Aleksandr-biochem** [MikkelDA/COBY#25]
I ran into

```
File ~/Documents/Porin_simulations/mdenv/lib/python3.10/site-packages/COBY/main_class/definition_preprocessors/lipid_scaffolds_preprocessor.py:29, in lipid_scaffolds_preprocessor.lipid_scaffolds_preprocessor(self)
     26         lipid_type_charges = tuple([tuple(details) for details in lipid_type_dict["charges"]])
     27     else:
     28         ### Converts to nested tuples
---> 29         if type(lipid_details) == list:
     30             lipid_type_charges = (tuple(lipid_type_dict["charges"]),)
     31 else:

UnboundLocalError: local variable 'lipid_details' referenced before assignment
```

The executed code:
```
sysname = "bilayer_coby"
outdir = os.path.join("simulations", sysname)
os.makedirs(outdir, exist_ok=True)

COBY.COBY(
    ### Designates box size
    box = [14.3, 14.3, 10], # [nm]
    
    ### " ".join([]) is used to turn the list of strings into a single string. Included for readability.
    ### The subargument "leaflet" is used to designate the currently targeted leaflet.
    ### The numbers after the lipids indicate their respective ratios within the specific leaflet
    
    membrane = " ".join([
        ### Subarguments initially target the entire membrane (e.g. both leaflets)
        ### Here we set the lipid parameter library to be used.
        ### It is "default" by default, so this is just to show how arguments affect the membrane/lealfets
        #"params:default",
        
        ### The upper leaflet is desgnated as the target for arguments using "leaflet:upper"
        ### Following arguments only affect that specific leaflet
        "leaflet:upper lipid:POPC", 

        ### The lower leaflet is desgnated as the target for arguments using "leaflet:lower"
        ### Following arguments only affect that specific leaflet        
        "leaflet:lower lipid:POPE:65 lipid:POPS:35",
    
    ]),
    
    ### 'solvation' argument solvates the system using the default solvation settings of "solv:W pos:NA neg:CL"
    ### The default concentration of water is 55.56 [mol/L] ("solv_molarity:55.56")
    ### The default salt concentration is 0.15 [mol/L] ("salt_molarity:0.15")
    solvation = "default",
    
    ### Uses topologies found in "toppar/top_for_COBY.itp" for charge determination
    itp_input = [
        "file:path_to/martini_v2.2.itp",
        "include:path/martini_v2.0_POPC_02.itp",
        "include:path/martini_v2.0_POPE_02.itp",
        "include:path/martini_v2.0_POPS_02.itp",
        "include:path/martini_v2.0_ions.itp",
    ],
    
    ### File writing
    out_sys = os.path.join(outdir, "_".join(["output", sysname])),
    out_top = os.path.join(outdir, "_".join(["topol",  sysname]) + ".top"),
    out_log = os.path.join(outdir, "_".join(["log",    sysname]) + ".log"),
    
    ### Designates the system name that is written in .pdb, .gro and .top files
    sn = sysname,
)
```

**Comment by MikkelDA** [comment]
Hello Aleksandr
Sorry for the delay. I was out of office the last few days, but i have looked at it now. It should be fixed in the next version (v0.2.3) which should be out sometime tomorrow.

**Comment by MikkelDA** [comment]
The update has now been pushed.

You can update COBY using:
`python -m pip install -U COBY`

Thanks for the bug report.
